### PR TITLE
azure_rm_sqldatabase: parse datetime module arguments

### LINKED ISSUE #619: Can't use restore_point_in_time in azure_rm_sqldatabase module
##### SUMMARY
We are trying to do PITR recovery of Azure SQL database but we are not able to format restore_point_in_time argument. We tried:

restore_point_in_time:  "2021-08-26 11:35:00"
restore_point_in_time:  "{{ '2021-08-26 11:35:00' | to_datetime }}"
restore_point_in_time:  "{{ ('2021-08-26 11:35:00' | to_datetime).isoformat() }}"

None of this were working

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure.azcollection.azure_rm_sqldatabase 

##### ANSIBLE VERSION

```
ansible [core 2.11.4]
  config file = None
  configured module search path = ['/home/wohnout/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  ansible collection location = /home/wohnout/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.10 (default, Jun  2 2021, 10:49:15) [GCC 9.4.0]
  jinja version = 2.10.1
  libyaml = True
```

##### COLLECTION VERSION

```
# /home/wohnout/.ansible/collections/ansible_collections
Collection         Version
------------------ -------
azure.azcollection 1.8.0
```

##### CONFIGURATION

```
Empty
```

##### OS / ENVIRONMENT

Called from ubuntu 20.04 to Azure

##### STEPS TO REPRODUCE

```yaml
- name: Creating PITR database
  azure_rm_sqldatabase:
    resource_group: "{{ azure_resource_group }}"
    create_mode: point_in_time_restore
    restore_point_in_time: "{{ ('2021-08-27 05:35:00' | to_datetime).isoformat() }}"
    source_database_id: "/subscriptions/{{ azure_subscription }}/resourceGroups/{{ azure_resource_group }}/providers/Microsoft.Sql/servers/{{ db_name }}/databases/test"
    server_name: "{{ db_name }}"
    name: test_PITR
    location: "{{ azure_resource_region }}"
```

##### EXPECTED RESULTS

task correctly processe

##### ACTUAL RESULTS

```paste below
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: wohnout
<127.0.0.1> EXEC /bin/sh -c 'echo ~wohnout && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/wohnout/.ansible/tmp `"&& mkdir "` echo /home/wohnout/.ansible/tmp/ansible-tmp-1630051202.1023154-395-137801296321864 `" && echo ansible-tmp-1630051202.1023154-395-137801296321864="` echo /home/wohnout/.ansible/tmp/ansible-tmp-1630051202.1023154-395-137801296321864 `" ) && sleep 0'
redirecting (type: modules) ansible.builtin.azure_rm_sqldatabase to azure.azcollection.azure_rm_sqldatabase
Using module file /home/wohnout/.ansible/collections/ansible_collections/azure/azcollection/plugins/modules/azure_rm_sqldatabase.py
<127.0.0.1> PUT /home/wohnout/.ansible/tmp/ansible-local-3902bclxy18/tmppdnf0zrx TO /home/wohnout/.ansible/tmp/ansible-tmp-1630051202.1023154-395-137801296321864/AnsiballZ_azure_rm_sqldatabase.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/wohnout/.ansible/tmp/ansible-tmp-1630051202.1023154-395-137801296321864/ /home/wohnout/.ansible/tmp/ansible-tmp-1630051202.1023154-395-137801296321864/AnsiballZ_azure_rm_sqldatabase.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python3 /home/wohnout/.ansible/tmp/ansible-tmp-1630051202.1023154-395-137801296321864/AnsiballZ_azure_rm_sqldatabase.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /home/wohnout/.ansible/tmp/ansible-tmp-1630051202.1023154-395-137801296321864/ > /dev/null 2>&1 && sleep 0'
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "ad_user": null,
            "adfs_authority_url": null,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": "auto",
            "cert_validation_mode": null,
            "client_id": null,
            "cloud_environment": "AzureCloud",
            "collation": null,
            "create_mode": "point_in_time_restore",
            "edition": null,
            "elastic_pool_name": null,
            "force_update": null,
            "location": "northeurope",
            "log_mode": null,
            "log_path": null,
            "max_size_bytes": null,
            "name": "test_PITR",
            "password": null,
            "profile": null,
            "read_scale": false,
            "recovery_services_recovery_point_resource_id": null,
            "resource_group": "resource-group",
            "restore_point_in_time": "2021-08-26T11:35:00",
            "sample_name": null,
            "secret": null,
            "server_name": "db-server-name",
            "sku": null,
            "source_database_deletion_date": null,
            "source_database_id": "/subscriptions/azure-subscription/resourceGroups/resource-group/providers/Microsoft.Sql/servers/db-server-name/databases/test",
            "state": "present",
            "subscription_id": null,
            "tags": null,
            "tenant": null,
            "zone_redundant": false
        }
    },
    "msg": "argument 'restore_point_in_time' is of type <class 'str'> and we were unable to convert to datetime: 'NoneType' object is not callable"
}
```

### PR DESCRIPTION
**Pull request by nbr23**
##### SUMMARY

In `azure_rm_sqldatabase`, the arguments `restore_point_in_time` and `source_database_deletion_date` are dates.
This bugfix parses the string passed as argument, and creates the datetime object required.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

- azure_rm_sqldatabase

##### ADDITIONAL INFORMATION

Currently the module expects `datetime` objtects to be passed directly as module arguments from the playbook. However, it appears that `argument_spec` does not handle `datetime` as an input type: https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#argument-spec

To correct this, this fix changed the type of `restore_point_in_time` and `source_database_deletion_date` to `str`, and calls dateutil.parse on the string to get the datetime object expected by the azure sdk.

This fixes https://github.com/ansible-collections/azure/issues/619

### CODE REVIEW

**Comment by Fred-sun** [comment]
tests/sanity/ignore-2.10.txt:273:1: A100: Ignoring 'invalid-ansiblemodule-schema' on 'plugins/modules/azure_rm_sqldatabase.py' is unnecessary

**Comment by nbr23** [comment]
> tests/sanity/ignore-2.10.txt:273:1: A100: Ignoring 'invalid-ansiblemodule-schema' on 'plugins/modules/azure_rm_sqldatabase.py' is unnecessary

Removed, thanks!

**Comment by Fred-sun** [comment]
> tests/sanity/ignore-2.10.txt:273:1: A100: Ignoring 'invalid-ansiblemodule-schema' on 'plugins/modules/azure_rm_sqldatabase.py' is unnecessary


Same operation (tests/sanity/ignore-2.11.txt and tests/sanity/ignore-2.12.txt)

**Comment by Fred-sun** [comment]
@nbr23 Add the follow line to ado.sh, Add python3-dateutil package!
```e.g.
     sudo apt install software-properties-common
     sudo add-apt-repository ppa:deadsnakes/ppa
     sudo apt install python3-dateutil
     sudo apt install python"$2" -y
```

https://github.com/nbr23/azure/blob/bed2d60f2b2dad7b1c82f845793c50403cfb9da2/tests/utils/ado/ado.sh#L19

File: plugins/modules/azure_rm_sqldatabase.py [hunk 1]
```diff
@@ -224,6 +224,7 @@
 
 import time
 from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase, format_resource_id
+import dateutil.parser
 
 try:
     from msrestazure.azure_exceptions import CloudError
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_sqldatabase.py:230]: @nbr23 Please move the imported dateutil to the location below. Thank you very much!
```suggestion
    import dateutil.parser
    from msrestazure.azure_exceptions import CloudError
```

File: plugins/modules/azure_rm_sqldatabase.py [hunk 2]
```diff
@@ -224,6 +224,7 @@
 
 import time
 from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase, format_resource_id
+import dateutil.parser
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_sqldatabase.py]:
```suggestion
```

**Comment by Fred-sun** [comment]
@nbr23 Can you add test cases for newly added logic? Thank you very much!

**Comment by nbr23** [comment]
@Fred-sun test case added. I had to add a fact to the returned value of sqldatabase_info so I can make sure there is a restore point (and get its date): https://github.com/ansible-collections/azure/pull/623/commits/ed903a901605a0a0bf82e3e1f55dde5ee24a3554

**Comment by xuzhang3** [comment]
LGTM

**Pull request merged**